src/main/java/uk/ac/ebi/eva/pipeline/io/readers/VcfHeaderReader.java

```diff
@@ -98,19 +98,33 @@ public VariantSourceEntity read() throws Exception {
             readAlreadyDone = true;
             return doRead();
         }
+
     }
 
-    private VariantSourceEntity doRead() {
+    private VariantSourceEntity doRead() throws IOException {
         if (variantReader == null) {
             throw new IllegalStateException("The method VcfHeaderReader.open() should be called before reading");
         }
         variantReader.pre();
         source.addMetadata(VARIANT_FILE_HEADER_KEY, variantReader.getHeader());
+
+        List<String> sampleNames = variantReader.getSampleNames();
+
+        Set<String> uniformSamples = new HashSet<String>();
```

Comment:
A shorter way would be to create the Set using the List as input, and comparing the sizes. To retrieve (possibly multiple) duplicate samples, you can use other methods in those classes.

I would also suggest renaming this variable to `uniqueSamples`, because that expresses the idea a bit better than `uniform`.